web/hooks/useReaderRef.ts

```diff
@@ -9,6 +9,41 @@ export const useReaderRef = ({
   onTableOfContents,
 }: Pick<ReadiumProps, 'file' | 'onLocationChange' | 'onTableOfContents' >) => {
   const readerRef = useRef<D2Reader | null>(null);
+  const [readingOrder, setReadingOrder] = useState<Locator[]>([]);
+
+  useEffect(() => {
```

Comment:
@matt5stones if reading order is a `ref` already, we can avoid converting it to a state object here and simply reference the `readerRef.current?.readingOrder` in the callback.